In [171]:
import math
import random

<h2>Count positive and negative numbers and compute the average of numbers</h2>Write a program that reads an unspecified number of integers, determines how many positive and negative values have been read, and computes the total and average of the input values(not counting zeros). Your program ends with the input 0. Display the average as a floating point number. Here is a sample run:<br><br>
Enter an integer, the input ends if it is 0: 1<br>
Enter an integer, the input ends if it is 0: 2<br>
Enter an integer, the input ends if it is 0: -1<br>
Enter an integer, the input ends if it is 0: 3<br>
Enter an integer, the input ends if it is 0: 0<br>
The number of positives is 3<br>
The number of negatives is 1<br>
The total is 5<br>
The average is 1.25<br>

In [13]:
#Prompt the user to enter a value 
userInput = int(input("Enter an integer, the input ends if it is 0: "))
total = 0
count = 0
numberPositives = 0
numberNegatives = 0
if userInput == 0:
    print("You didn't enter any number")
else:
    while userInput != 0:
        total = total + userInput
        count = count + 1
        if userInput > 0:
            numberPositives = numberPositives  + 1
        else:
            numberNegatives = numberNegatives + 1
        userInput = int(input("Enter an integer, the input ends if it is 0: "))
    print("The number of positives is",numberPositives)
    print("The number of negatives is",numberNegatives)
    print("The total is",total)
    print("The average is",total / count)

Enter an integer, the input ends if it is 0: 1
Enter an integer, the input ends if it is 0: 2
Enter an integer, the input ends if it is 0: -1
Enter an integer, the input ends if it is 0: 3
Enter an integer, the input ends if it is 0: 0
The number of positives is 3
The number of negatives is 1
The total is 5
The average is 1.25


<h2>Conversion from kilograms to pounds</h2>Write a program that displays the coversion between kilograms and pounds. 1 kilogram is 2.2 pounds. 

In [22]:
KILOGRAM_PER_POUNDS = 2.2

print("Kilograms","\t","Pounds")
for num in range(20):
    if num % 2 != 0:
        print(num, "\t\t",format(num*KILOGRAM_PER_POUNDS,".1f"))

Kilograms 	 Pounds
1 		 2.2
3 		 6.6
5 		 11.0
7 		 15.4
9 		 19.8
11 		 24.2
13 		 28.6
15 		 33.0
17 		 37.4
19 		 41.8


<h2>Conversion from miles to kilometers</h2>Write a program that displays the coversion between kilograms and pounds. 1 mile is 1.609 kilometers. 

In [27]:
MILE_PER_KILOMETERS = 1.609   #Constant

print("Miles","\t","Kilometers")
for num in range(1,11):
    print(num, "\t",format(num*MILE_PER_KILOMETERS,".3f"))

Miles 	 Kilometers
1 	 1.609
2 	 3.218
3 	 4.827
4 	 6.436
5 	 8.045
6 	 9.654
7 	 11.263
8 	 12.872
9 	 14.481
10 	 16.090


<h2>Conversion from kilograms to pounds and pounds to kilograms</h2>Write a program that displays the conversion from kilograms to pounds and pounds to kilograms. 

In [161]:
KILOGRAM_PER_POUNDS = 2.2
POUNDS_PER_KILOGRAM = 0.4545
startPound = 20
print("Kilograms","\t","Pounds", "\t","|", "\t", "Pounds","\t","Kilograms")
for num in range(1,20):
    if num % 2 != 0:
        print(num,"\t\t",format(num*KILOGRAM_PER_POUNDS,".1f"), "\t\t","|", "\t", end = " ")
        print(startPound,"\t\t",format(startPound*POUNDS_PER_KILOGRAM,".2f"))
        startPound = startPound + 5

Kilograms 	 Pounds 	 | 	 Pounds 	 Kilograms
1 		 2.2 		 | 	 20 		 9.09
3 		 6.6 		 | 	 25 		 11.36
5 		 11.0 		 | 	 30 		 13.63
7 		 15.4 		 | 	 35 		 15.91
9 		 19.8 		 | 	 40 		 18.18
11 		 24.2 		 | 	 45 		 20.45
13 		 28.6 		 | 	 50 		 22.73
15 		 33.0 		 | 	 55 		 25.00
17 		 37.4 		 | 	 60 		 27.27
19 		 41.8 		 | 	 65 		 29.54


<h2>Use trigonometric functions</h2>Print the a table to display the sin value and cos value of degrees from 0 to 360 with increments of 10 degrees. Round the value to keep four digits after the decimal point.

In [81]:
#Display the 
print("Degree", "\t\t","Sin","\t\t","Cos")

for number in range(0, 75,10):
    radiansNum = math.radians(number)
    print(number, "\t\t", format(math.sin(radiansNum),".4f"),"\t", format(math.cos(radiansNum),".4f"))

Degree 		 Sin 		 Cos
0 		 0.0000 	 1.0000
10 		 0.1736 	 0.9848
20 		 0.3420 	 0.9397
30 		 0.5000 	 0.8660
40 		 0.6428 	 0.7660
50 		 0.7660 	 0.6428
60 		 0.8660 	 0.5000
70 		 0.9397 	 0.3420


<h2>Use the math.sqrt function</h2>Write a program that prints the square root of the numbers 1 to 11 using the math.sqrt function

In [85]:
#Display the header
print("NUmber", "\t\t","Square Root")

#Display the square of the numbers
for number in range(0,11):
    print(number, "\t\t", format(math.sqrt(number), ".4f"))

NUmber 		 Square Root
0 		 0.0000
1 		 1.0000
2 		 1.4142
3 		 1.7321
4 		 2.0000
5 		 2.2361
6 		 2.4495
7 		 2.6458
8 		 2.8284
9 		 3.0000
10 		 3.1623


<h2>Find the smallest n such that n<sup>2</sup> &gt; 12000</h2>Use a while loop to find the smallest integer n such that n<sup>2</sup> is greater than 12000

In [86]:
n = 1
while n*n < 12000:
    n = n + 1
print(n)

110


<h2>Find the largest n such that n<sup>3</sup> &lt; 12000</h2>Use a while loop to find the largest integer n such that n<sup>3</sup> is less than 12000

In [95]:
p = 1
while 12000 > pow(p,3):
    p = p + 1

if p*p*p > 12000:
    print(p-1)
else:
    print(p)

22


<h2>Financial application: compare loans with various interest rates</h2>Write a program that lets the user enter the loan amount and the loan period in number of years and displays the monthly and total payments for each interest rate starting from 5% to 8%, with an increment of 1/8. Here is a sample run:<br>

In [109]:
#Prompt the user to enter the loan amount
loanAmount = int(input("Enter Loan Amount: "))

#Prompt the user to enter the number of years
numberYears = int(input("Enter number of years: "))

print("Interest Rate","\t", "Monthly Payment", "\t", "Total Payment")

for number in range(0,25):
    annualInterest = 5 + number * 0.125
    monthlyInterestRate = annualInterest / 1200
    monthlyPayment = loanAmount * monthlyInterestRate / (1 - 1 / (1 + monthlyInterestRate)**(numberYears * 12))
    totalPayment = monthlyPayment * numberYears * 12
    print(format(annualInterest,".3f"),"%", "\t", format(monthlyPayment, ".2f"), "\t\t",format(totalPayment,".2f"))

Enter Loan Amount: 10000
Enter number of years: 5
Interest Rate 	 Monthly Payment 	 Total Payment
5.000 % 	 188.71 		 11322.74
5.125 % 	 189.29 		 11357.13
5.250 % 	 189.86 		 11391.59
5.375 % 	 190.44 		 11426.11
5.500 % 	 191.01 		 11460.70
5.625 % 	 191.59 		 11495.35
5.750 % 	 192.17 		 11530.06
5.875 % 	 192.75 		 11564.84
6.000 % 	 193.33 		 11599.68
6.125 % 	 193.91 		 11634.59
6.250 % 	 194.49 		 11669.56
6.375 % 	 195.08 		 11704.59
6.500 % 	 195.66 		 11739.69
6.625 % 	 196.25 		 11774.85
6.750 % 	 196.83 		 11810.08
6.875 % 	 197.42 		 11845.37
7.000 % 	 198.01 		 11880.72
7.125 % 	 198.60 		 11916.14
7.250 % 	 199.19 		 11951.62
7.375 % 	 199.79 		 11987.16
7.500 % 	 200.38 		 12022.77
7.625 % 	 200.97 		 12058.44
7.750 % 	 201.57 		 12094.18
7.875 % 	 202.17 		 12129.97
8.000 % 	 202.76 		 12165.84


<h2>Financial Application: loan amortization schedule</h2>The monthly payment for a given loan pays the principal and interest. The monthly interest is computed by multiplying the monthly interest rate and the balance (the remaining principal). The principal paid for the month is therefore the monthly payment minus the monthly interest. Write a program that lets the user enter the loan amount, number of years, and interest rate, and then displays the amortization schedule for loan. 

In [137]:
#Prompt the user to enter the loan amount
amountLoan = int(input("Enter the Loan Amount: "))

#Prompt the user to enter the number of years
numberOfYears = int(input("Enter the number of Years: "))

#Prompt the user to enter the annual interest rate
annualInterestRate = int(input("Enter the annual Interest Rate: "))

#Compute the monthly interest rate
monthlyInterestRate = annualInterestRate / 1200

#Compute the monthly payment rate and the total payment
monthlyPaymentRate = amountLoan * monthlyInterestRate / (1 - 1 / (1 + monthlyInterestRate)**(numberOfYears * 12))
totalPayment = monthlyPaymentRate * numberOfYears * 12

print("Monthly Payment: ", format(monthlyPaymentRate, ".2f"))
print("Total Payment: ", format(totalPayment, ".2f"))
print("Payment#", "\t", "Interest", "\t", "Principal","\t", "Balance")
balance = amountLoan



for num in range(1, numberOfYears*12 + 1):
    interest = monthlyInterestRate * balance
    principal = monthlyPaymentRate - interest
    balance = balance - principal
    print(num, "\t\t", format(interest,".2f"), "\t\t", format(principal,".2f"), "\t", format(balance, ".2f"))

Enter the Loan Amount: 10000
Enter the number of Years: 1
Enter the annual Interest Rate: 7
Monthly Payment:  865.27
Total Payment:  10383.21
Payment# 	 Interest 	 Principal 	 Balance
1 		 58.33 		 806.93 	 9193.07
2 		 53.63 		 811.64 	 8381.42
3 		 48.89 		 816.38 	 7565.05
4 		 44.13 		 821.14 	 6743.91
5 		 39.34 		 825.93 	 5917.98
6 		 34.52 		 830.75 	 5087.24
7 		 29.68 		 835.59 	 4251.65
8 		 24.80 		 840.47 	 3411.18
9 		 19.90 		 845.37 	 2565.81
10 		 14.97 		 850.30 	 1715.51
11 		 10.01 		 855.26 	 860.25
12 		 5.02 		 860.25 	 0.00


<h2>Sum of a series</h2> Write a program to sum the following series:
$\frac{1}{3} + \frac{3}{5} + \frac{5}{7} + \frac{7}{9} + \frac{9}{11} + \frac{11}{13}  + ... +  \frac{95}{97} + \frac{97}{99}$

In [136]:
#Initialize the total variable
total = 0

for number in range(1,98,2):
    total = total + ( number / number + 2)
    
#Display the result    
print(format(total, ".2f"))

147.00


In [153]:
totalNumbers = 0
count = 0
for number in range(1,8,1):
    num = 0
    num = number + num
    while num < 8:
        print(number,num)
        num = num + 1
        count = count + 1
print(count)

1 1
1 2
1 3
1 4
1 5
1 6
1 7
2 2
2 3
2 4
2 5
2 6
2 7
3 3
3 4
3 5
3 6
3 7
4 4
4 5
4 6
4 7
5 5
5 6
5 7
6 6
6 7
7 7
28


<h2>Financial application: compound value</h2>Suppose you save <span>&dollar;</span>100 each month into a savings account with the annual interest rate 5%. So, the monthly interest rate is 0.05712 = 0.00417. After the first month, the value in the account becomes<br>
$100 * (1 + 0.00417) = 100.417$<br>
After the second month, the value in the account becomes <br>
$(100 + 100.417) * (1 + 0.00417) = 201.252$<br>
After the third month, the value in the account becomes <br>
$(100 + 201.252) * (1 + 0.00417) = 302.507$<br> 
and so on. Write a program that prompts the user to enter an amount (e.g.,100), the annual interest rate rate (e.g.,5), and the number of months (e.g.,6) and displays the amount in the savings account after the given month

In [159]:
#Prompt the user to enter the amount
savingAmount = float(input("Enter the savings amount: "))

#Prompt the user to enter the annual interest rate
annualInterestRate = float(input("Enter the annual interest rate: "))
monthlyInterestRate = annualInterestRate / 1200
totalAmount = 0 

#Prompt the user to enter the number of months 
numberOfMonths = int(input("Enter the number of months: "))

for number in range(numberOfMonths):
    totalAmount = (savingAmount + totalAmount) * (1 + monthlyInterestRate) 
    
    
print("The amount in the Savings Amount after", numberOfMonths, "months is", format(totalAmount, ".3f"))

Enter the savings amount: 100
Enter the annual interest rate: 5
Enter the number of months: 3
The amount in the Savings Amount after 3 months is 302.507


<h2>Financial application: compute CD value</h2>Suppose you put <span>&dollar;</span>10,000 into a CD with an annual percentage yield of 5.75%. After one month, the CD is worth<br>
$10000 + 10000* 5.75 / 1200 = 10047.91$ <br>
After two months, the CD is worth<br>
$10047.91 + 10047.91 * 5.75 / 1200 = 10096.06$<br>
After three months, the CD is worth<br>
$10096.06 + 10096.06 * 5.75 / 1200 = 10145.43$<br>
and so on. Write a program that prompts the user to enter an amount (e.g., 10000), the annual percentage yield (e.g.,5.75), and the number of months (e.g.,18), and displays a table with the values.

In [170]:
#Prompt the user to enter the initial deposit amount
initialDepositAmount = float(input("Enter the initial deposit amount: "))

#Prompt the user to enter the annual yield percentage
yieldPercentage = float(input("Enter annual percentage yield: "))

#Prompt the user to enter the maturity period
maturityPeriod = int(input("Enter maturity period(number of months): "))

totalAmountYielded = initialDepositAmount
monthlyInterestRate = yieldPercentage / 1200

#Print the header
print("Month","\t", "CD Value")

for number in range(1,maturityPeriod+1):
    totalAmountYielded = totalAmountYielded + (totalAmountYielded * monthlyInterestRate) 
    print(number,"\t", format(totalAmountYielded, ".2f"))

Enter the initial deposit amount: 10000
Enter annual percentage yield: 5.75
Enter maturity period(number of months): 18
Month 	 CD Value
1 	 10047.92
2 	 10096.06
3 	 10144.44
4 	 10193.05
5 	 10241.89
6 	 10290.97
7 	 10340.28
8 	 10389.82
9 	 10439.61
10 	 10489.63
11 	 10539.89
12 	 10590.40
13 	 10641.14
14 	 10692.13
15 	 10743.37
16 	 10794.84
17 	 10846.57
18 	 10898.54


In [174]:
summedTotal = 0

for number in range(9):
    summedTotal = summedTotal + 1 / (math.sqrt(number) + math.sqrt(number + 1))
    
print(summedTotal)

3.0
